src/api/spec/helpers/webui/build_result_helper.rb

```diff
@@ -0,0 +1,99 @@
+require 'rails_helper'
+
+RSpec.describe Webui::BuildresultHelper do
+  include Webui::WebuiHelper
+  let(:project) { create(:project) }
+  let(:package) { create(:package, project: project) }
+  let(:arch) { 'i586' }
+  let(:repo) { create(:repository, project: project, architectures: [arch]) }
+
+  describe '#arch_repo_table_cell' do
+    let(:description) { Buildresult::STATUS_DESCRIPTION[key] }
+    let(:status) { { "code" => key, "details" => description } }
+
+    before do
+      allow(helper).to receive(:valid_xml_id) do |rawid|
+        rawid
+      end
+      allow(helper).to receive(:sprite_tag) do |icon, opts|
+        sprite_tag(icon, opts)
+      end
+
+      assign(:statushash, statushash)
+      assign(:project, project)
+    end
+
+    RSpec.shared_examples "generic case" do
+      context "without status and statushash" do
+        let(:statushash) { { repo.name => { arch => { package.name => { "code" => key, "details" => description } } } } }
+
+        it { expect(helper.arch_repo_table_cell(repo.name, arch, package.name)).to eq result }
+      end
+
+      context "with status and without enable_help" do
+        let(:hash_key) { key == "scheduled" ? "unknown" : "scheduled" } # to have different key in the statushash
+        let(:statushash) { { repo.name => { arch => { package.name => { "code" => hash_key, "details" => description } } } } }
+
+        it { expect(helper.arch_repo_table_cell(repo.name, arch, package.name, status, false)).to eq result2 }
+      end
+    end
+
+    %w(succeeded failed broken dispatching building signing finished disabled locked unknown).each do |key|
+      context "with #{key}" do
+        let(:key) {key}
+        let(:encoded_description) { description.gsub("'", "&#39;") } # ' is encoded as &#39;
+        let(:result) do
+          "<td class=\"status_#{key} buildstatus nowrap\"><a title=\"#{encoded_description}\" rel=\"nofollow\" " +
```

Comment:
Because it would look like [this](https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/models/project/update_from_xml_command_spec.rb#L15):

```ruby
       xml_hash = Xmlhash.parse(
          <<-EOF
            <project name="#{project.name}">
              <repository name="repo_1" />
              <repository name="new_repo" rebuild="local" block="never" linkedbuild="all" />
            </project>
          EOF
        )
```